Source/Core/Core/NetPlayClient.cpp

```diff
@@ -553,6 +569,216 @@ unsigned int NetPlayClient::OnData(sf::Packet& packet)
+
+    case SYNC_SAVE_DATA_RAW:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      bool is_slot_a;
+      std::string region;
+      bool mc251;
+      packet >> is_slot_a >> region >> mc251;
+
+      const std::string path = File::GetUserPath(D_GCUSER_IDX) + GC_MEMCARD_NETPLAY +
+                               (is_slot_a ? "A." : "B.") + region + (mc251 ? ".251" : "") + ".raw";
+      if (File::Exists(path) && !File::Delete(path))
+      {
+        PanicAlertT("Failed to delete NetPlay memory card. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      const bool success = DecompressPacketIntoFile(packet, path);
+      SyncSaveDataResponse(success);
+    }
+    break;
+
+    case SYNC_SAVE_DATA_GCI:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      bool is_slot_a;
+      u8 file_count;
+      packet >> is_slot_a >> file_count;
+
+      const std::string path = File::GetUserPath(D_GCUSER_IDX) + GC_MEMCARD_NETPLAY DIR_SEP +
+                               StringFromFormat("Card %c", is_slot_a ? 'A' : 'B');
+
+      if ((File::Exists(path) && !File::DeleteDirRecursively(path + DIR_SEP)) ||
+          !File::CreateFullPath(path + DIR_SEP))
+      {
+        PanicAlertT("Failed to reset NetPlay GCI folder. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      for (u8 i = 0; i < file_count; i++)
+      {
+        std::string file_name;
+        packet >> file_name;
+
+        if (!DecompressPacketIntoFile(packet, path + DIR_SEP + file_name))
+        {
+          SyncSaveDataResponse(false);
+          return 0;
+        }
+      }
+
+      SyncSaveDataResponse(true);
+    }
+    break;
+
+    case SYNC_SAVE_DATA_WII:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      const auto game = m_dialog->FindGameFile(m_selected_game);
+      if (game == nullptr)
+      {
+        SyncSaveDataResponse(true);  // whatever, we won't be booting anyways
+        return 0;
+      }
+
+      const std::string path = File::GetUserPath(D_USER_IDX) + "Wii" GC_MEMCARD_NETPLAY DIR_SEP;
+
+      if (File::Exists(path) && !File::DeleteDirRecursively(path))
+      {
+        PanicAlertT("Failed to reset NetPlay NAND folder. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      auto temp_fs = std::make_unique<IOS::HLE::FS::HostFileSystem>(path);
+      temp_fs->CreateDirectory(IOS::PID_KERNEL, IOS::PID_KERNEL,
+                               Common::GetTitleDataPath(game->GetTitleID()), 0,
+                               {IOS::HLE::FS::Mode::ReadWrite, IOS::HLE::FS::Mode::ReadWrite,
+                                IOS::HLE::FS::Mode::ReadWrite});
+      auto save = WiiSave::MakeNandStorage(temp_fs.get(), game->GetTitleID());
+
+      bool exists;
+      packet >> exists;
+      if (exists)
+      {
+        // Stupid temporaries because of BigEndianValue
```

Comment:
Couldn't we simply add overloads for `operator>>` (and maybe `operator<<` for symmetry, if necessary) to do this for us on `BigEndianValue`?

Reply:
It would need an overload that handles `sf::Packet`, which is rather specialized and I'm not sure we want to drag SFML headers into everything that uses `BigEndianValue`.

Reply:
Couldn't they be next to their usage in this file?
Then again, I didn't think about that, so I'm ok with not doing it too. Not worth pulling all that stuff in for that.

Reply:
Alright, I've come up with a decent solution for this, and also cleaned up handling of 64-bit types in `sf::Packet` along with it.